components/tools/src/main/java/edu/jhuapl/dorset/components/tools/CommandLineClient.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * 
+ */
+package edu.jhuapl.dorset.components.tools;
+
+import java.util.Scanner;
+
+import edu.jhuapl.dorset.Application;
+import edu.jhuapl.dorset.Request;
+import edu.jhuapl.dorset.Response;
+import edu.jhuapl.dorset.agents.AbstractAgent;
+import edu.jhuapl.dorset.routing.Router;
+import edu.jhuapl.dorset.routing.SingleAgentRouter;
+
+/**
+ * Simple, command line access to a single agent, router, or application.
+ * Whatever you enter at the command line is sent, and the text response is
+ * provided to System.out.
+ * 
+ * The default character to exit the system is 'q'.
+ * 
+ * Example:
+ * 
+ * <pre>
+ * Agent agent = new CalculatorAgent();
+ * CommandLineClient client = new CommandLineClient(agent);
+ * client.go();
+ * </pre>
+ * 
+ * @author david.patrone
```

Comment:
We don't want @author comments since for most files its likely that multiple people will touch them and we can maintain who did what through the git history.